manifests/sssd/client.pp

```diff
@@ -38,40 +43,49 @@
 # @param min_id
 #   The lowest user ID that SSSD should recognize from the remote server
 #
-# @author Trevor Vaughan <tvaughan@onyxpoint.com>
+# @author https://github.com/simp/pupmod-simp-simp/graphs/contributors
 #
 class simp::sssd::client (
   Boolean $ldap_domain       = simplib::lookup('simp_options::ldap', { 'default_value' => false }),
-  Boolean $local_domain      = true,
-  Boolean $autofs            = true,
-  Boolean $sudo              = true,
-  Boolean $ssh               = true,
+  Boolean $local_domain,     #data in module
+  Boolean $autofs            = true, #deprecated
+  Boolean $sudo              = true, #deprecated
+  Boolean $ssh               = true, #deprecated
   Boolean $enumerate_users   = false,
   Boolean $cache_credentials = true,
   Integer $min_id            = 500
 ){
 
   simplib::module_metadata::assert($module_name, { 'blacklist' => ['Windows'] })
 
-  if $ldap_domain or $local_domain {
-    include 'sssd'
-    include 'sssd::service::nss'
-    include 'sssd::service::pam'
+  # Don't attemt to setup sssd in el6 or 7 if a local or ldap domain is not defined.
 
-    if $autofs { include 'sssd::service::autofs' }
-    if $sudo { include 'sssd::service::sudo' }
-    if $ssh { include 'sssd::service::ssh' }
+  if $local_domain or $ldap_domain or versioncmp($facts['os']['release']['major'], '8') >= 0 {
```

Comment:
I'm not sure why you need the `versioncmp` here if these parameters are controlled by the module data.

Reply:
Because you can set up sssd without a domain in EL8 and it works.  It is a recommended configuration in the docs I read because "they say" it speeds up authentication because of how it stores the data.